imap_data_access/io.py

```diff
@@ -159,6 +162,54 @@ def query(
             "At least one query parameter must be provided. "
             "Run 'query -h' for more information."
         )
+    # Check instrument name
+    if instrument is not None and instrument not in imap_data_access.VALID_INSTRUMENTS:
+        raise ValueError(
+            "Not a valid instrument, please choose from "
+            + ", ".join(imap_data_access.VALID_INSTRUMENTS)
+        )
+
+    # Check data-level
+    # do an if statement that checks that data_level was passed in,
+    # then check it against all options, l0, l1a, l1b, l2, l3 etc.
+    if data_level is not None and data_level not in imap_data_access.VALID_DATALEVELS:
+        raise ValueError(
+            "Not a valid data level, choose from "
+            + ", ".join(imap_data_access.VALID_DATALEVELS)
+        )
+
+    # Check start-date
+    if start_date is not None and not file_validation.ScienceFilePath.is_valid_date(
+        start_date
+    ):
+        raise ValueError("Not a valid start date, use format 'YYYYMMDD'.")
+
+    # Check end-date
+    if end_date is not None and not file_validation.ScienceFilePath.is_valid_date(
+        end_date
+    ):
+        raise ValueError("Not a valid end date, use format 'YYYYMMDD'.")
+
+    # Check version make sure to include 'latest'
+    if version is not None and not file_validation.ScienceFilePath.is_valid_version(
+        version
+    ):
+        raise ValueError("Not a valid version, use format 'vXXX'.")
+
+    # check repointing follows 'repoint00000' format
+    if (
+        repointing is not None
+        and not file_validation.ScienceFilePath.is_valid_repointing(repointing)
+    ):
+        raise ValueError(
+            "Not a valid repointing, use format repointing<num>,"
```

Comment:
Remove the `ing` and make it explicit how many numbers there has to be. (Update the actual code as well, I just noticed this in the test here)
```suggestion
            "Not a valid repointing, use format repointXXXXX,"
```